src/renderer/electron-types.ts

```diff
@@ -19,41 +24,84 @@ export class ElectronTypes {
 
   constructor(
     private readonly monaco: typeof MonacoType,
-    private readonly cacheDir: string,
+    private readonly electronCacheDir: string,
+    private readonly nodeCacheDir: string,
   ) {}
 
   public async setVersion(ver?: RunnableVersion): Promise<void> {
     this.clear();
 
     if (!ver) return;
+
+    await this.setElectronTypes(ver);
+    await this.setNodeTypes(ver.version);
+  }
+
+  public async setElectronTypes(ver: RunnableVersion): Promise<void> {
     const { localPath: dir, source, version } = ver;
 
+    // If it's a local development version, pull Electron types from out directory.
     if (dir) {
       const file = path.join(dir, 'gen/electron/tsc/typings', ELECTRON_DTS);
-      this.setTypesFromFile(file, ver);
+      this.setTypesFromFile(file, ver.version);
```

Comment:
_nit_ `version` already extracted on line 41

```suggestion
      this.setTypesFromFile(file, version);
```

Same thing below on 49, 58, 60